contracts/zero-ex/contracts/src/external/Puppet.sol

```diff
@@ -0,0 +1,175 @@
+/*
+
+  Copyright 2020 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.6.5;
+pragma experimental ABIEncoderV2;
+
+import "@0x/contracts-utils/contracts/src/v06/errors/LibRichErrorsV06.sol";
+import "@0x/contracts-utils/contracts/src/v06/errors/LibOwnableRichErrorsV06.sol";
+import "../errors/LibPuppetRichErrors.sol";
+import "./IPuppet.sol";
+
+
+/// @dev A contract that can execute arbitrary calls from its owner.
+contract Puppet is
```

Comment:
I'm curious to hear what other protocologists think, but the terminology has been tripping me up a bit. It's definitely clever, but I wonder if it's really necessary. The puppet and token spender are basically just implementing the proxy pattern. I might consider more familiar (albeit boring) names like `CallProxy` and `UniversalAssetProxy`.

Reply:
Technically the puppet can do a regular call as well. We don't use it right now though so we might as well drop that feature and go with your naming convention.

Reply:
I kinda prefer FlashWallet? (flash loans so hot right now)

Reply:
I like FlashWallet